### PR TITLE
Avoid PAL LoadLibrary simulator for regular PInvoke

### LINKED ISSUE #21009: PAL: Remove PAL registration for Native file handles
PAL API registers file handles to create a HMODULE structure from a NATIVE_FILE_HANDLE.
Investigate whether this process is necessary, and if not remove it.

One of the things that PAL registration does is to call DLLMain() on a native DLL to mimic Windows style functionality. However, this is likely unnecessary for Unix DLLs, and can be removed.

### PR DESCRIPTION
**Pull request by jkotas**
Fixes #21009

### CODE REVIEW

File: src/vm/util.hpp [hunk 1]
```diff
@@ -810,9 +810,18 @@ inline void UnsafeTlsFreeForHolder(DWORD* addr)
...
+
+#ifdef FEATURE_PAL
+    PAL_FreeLibraryDirect(h);
+#else
+    FreeLibrary(h);
+#endif
+}
 
-typedef Wrapper<HMODULE, DoNothing<HMODULE>, VoidFreeLibrary, NULL> ModuleHandleHolder;
+typedef Wrapper<HMODULE, DoNothing<NATIVE_LIBRARY_HANDLE>, VoidFreeNativeLibrary, NULL> NativeLibraryHandleHolder;
```
Comment (AaronRobinsonMSFT) [inline src/vm/util.hpp]: ~~`HMODULE`~~ -> `NATIVE_LIBRARY_HANDLE `

File: src/vm/appdomain.cpp [hunk 2]
```diff
@@ -4723,7 +4722,7 @@ void AppDomain::AddUnmanagedImageToCache(LPCWSTR libraryName, HMODULE hMod)
 }
 
 
-HMODULE AppDomain::FindUnmanagedImageInCache(LPCWSTR libraryName)
+NATIVE_LIBRARY_HANDLE AppDomain::FindUnmanagedImageInCache(LPCWSTR libraryName)
 {
     CONTRACT(HMODULE)
```
Comment (AaronRobinsonMSFT) [inline src/vm/appdomain.cpp]: ~~`HMODULE`~~ -> `NATIVE_LIBRARY_HANDLE`

File: src/vm/util.hpp [hunk 3]
```diff
@@ -810,9 +810,18 @@ inline void UnsafeTlsFreeForHolder(DWORD* addr)
 typedef Holder<DWORD*, DoNothing<DWORD*>, UnsafeTlsFreeForHolder> TlsHolder;
 
 // A holder for HMODULE.
-FORCEINLINE void VoidFreeLibrary(HMODULE h) { WRAPPER_NO_CONTRACT; CLRFreeLibrary(h); }
+FORCEINLINE void VoidFreeNativeLibrary(NATIVE_LIBRARY_HANDLE h)
```
Comment (am11) [inline src/vm/util.hpp]: Should we test `h` for NULL? `dlclose` segmentation faults when input is null: [`dlopen`/`dlerror`/`dlclose` experiment](https://wandbox.org/permlink/Ghm7YjwU4pBAlTT2).

Comment (jkotas) [reply src/vm/util.hpp]: Good catch. Thanks! https://github.com/dotnet/coreclr/pull/24685

**Pull request merged**